### PR TITLE
feat: single cache implementation

### PR DESCRIPTION
**Pull request by pgautier404**

### CODE REVIEW

File: .github/workflows/on-push-to-main.yml [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+
+env:
+  SIMPLETEST_DB: "mysql://drupal:drupal@127.0.0.1:3306/drupal"
+  SIMPLETEST_BASE_URL: "http://127.0.0.1:8080"
+  MODULE_FOLDER: "drupal-cache"
+  MODULE_REPO: "momentohq/drupal-cache"
+  DRUPAL_MODULE_NAME: "momento_cache"
+  DRUPAL_CORE_VERSION: 9.4.x
+  BRANCH_NAME: ${{ github.head_ref || github.ref_name }}
+  # MOMENTO_API_TOKEN: ${{ secrets.ALPHA_TEST_AUTH_TOKEN }}
```
Comment (pgautier404) [inline .github/workflows/on-push-to-main.yml:19]: Need to get this secret in place as well as `MOMENTO_MACHINE_USER_GITHUB_TOKEN`.

File: .github/workflows/on-push-to-main.yml [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+  MOMENTO_API_TOKEN: ${{ secrets.MOMENTO_API_TOKEN }}
+
+jobs:
+  main:
+    runs-on: ubuntu-latest
+    strategy:
+      max-parallel: 1
+      fail-fast: true
+      matrix:
+        php-version: ['7.4', '8.0', '8.1'] # , '8.2']
```
Comment (pgautier404) [inline .github/workflows/on-push-to-main.yml:29]: PHP 8.2 is incompatible with Drupal 9.4, which is what we're targeting right now. I'll expand testing to PHP 8.2 and Drupal 9.5 and 10 when I can.

File: .github/workflows/on-push-to-main.yml [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+        run: |
+          php -d sendmail_path=$(which true); vendor/bin/drush --yes -v \
+            site-install minimal --db-url="$SIMPLETEST_DB"
+          vendor/bin/drush en $DRUPAL_MODULE_NAME -y
+
+      - name: Run PHPCS
+        working-directory: ${{ env.DRUPAL_ROOT }}
+        run: |
+          # vendor/bin/phpcs $MODULE_FOLDER --standard=Drupal \
+          #  --extensions=php,module,inc,install,test,info
```
Comment (pgautier404) [inline .github/workflows/on-push-to-main.yml:128]: Linting. Another thing I'll need to look at when I have more time but is currently not impacting functionality.

File: .github/workflows/on-push-to-main.yml (hunk 1 above)
Comment (cprice404) [reply .github/workflows/on-push-to-main.yml:19]: these should both be done

**Review by cprice404 — APPROVED** [review]

**Pull request merged**